src/cascadia/TerminalSettingsModel/IconPathConverter.cpp

```diff
@@ -297,8 +297,12 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         if (commaIndex != std::wstring::npos)
         {
             // Convert the string iconIndex to an int
-            const auto index = til::to_ulong(pathView.substr(commaIndex + 1));
-            if (index == til::to_ulong_error)
+            const auto indexString = pathView.substr(commaIndex + 1);
+            std::wstringstream stringStream;
+            stringStream << indexString;
+            int index = 0;
+            stringStream >> index;
+            if (index == 0)

```

Comment:
Curious about this change here - this should have only been used for the WinUI icons within the app (and not the jumplist ones). Did this need to get changed to for some reason? Or was this merely trying to align the icon parsing code?

Reply:
So SetIconLocation takes an number that can be either positive or negative. positive is an index and negative is an ID. So if someone passes a negative number as an ID then the Icon would work in the jump list but not in the UI.

Reply:
TIL

Reply:
pushed some changes. Always open to feedback :)